src/main/java/org/cryptomator/common/keychain/KeychainManager.java

```diff
@@ -90,6 +93,8 @@ public boolean isLocked() {
 	 * @throws KeychainAccessException
 	 */
 	public boolean isPassphraseStored(String key) throws KeychainAccessException {
+		// check if keyrings are disabled; in this case we don't need to ask the backend
+		if ( !settings.allowAllKeyrings().get() ) return false;
```

Comment:
Is this really necessary? Shouldn't we rather modify `KeychainModule#provideKeychainAccessProvider()` in order to not contain any keychain if using them is disabled? That way any interaction with a keychain within this class will fail due to `getKeychainOrFail()` throwing an exception. We need to make sure though that any access to this method is guarded by previously checking whether the keychain is available.